### PR TITLE
[Rust Frontend] Add MiniMax M2 append-think reasoning parser

### PR DESCRIPTION
**Pull request by 982945902**
## Summary

- add the explicit `minimax_m2_append_think` Rust reasoning parser
- prepend `<think>` once while preserving all non-empty generated output as visible content
- keep an empty generated stream empty because the Rust parser is shared by streaming and collected responses
- preserve special tokens and keep automatic MiniMax model routing on the normal parser
- cover parser lifecycle, empty deltas and streams, factory registration, and parser-name diagnostics

## Duplicate-work check

This does not duplicate an existing PR. Before implementation I checked:

- roadmap issue #44280 and its comments
- open PRs referencing `44280`
- open PR searches for `minimax_m2_append_think` and MiniMax M2 append-think parser terms

No overlapping Rust implementation or claim was found. The item was claimed in #44280 before implementation.

## Tests

- `cargo test --manifest-path rust/Cargo.toml -p vllm-parser --lib` — 435 passed
- `cargo test --manifest-path rust/Cargo.toml -p vllm-chat --lib` — 279 passed
- `cargo clippy --manifest-path rust/Cargo.toml -p vllm-parser -p vllm-chat --all-targets -- -D warnings` — passed
- `uv tool run pre-commit run --files <changed files>` — passed

## Full-model serving regression

A CPU-only end-to-end serving regression was run with the locally built PR Rust binary and the full `allenai/Olmo-3-7B-Instruct` weights (13.59 GiB), using `--reasoning-parser minimax_m2_append_think`. A different model was intentionally used because the parser is selected explicitly and MiniMax M2.5 itself is approximately 434 GiB across 125 weight shards.

- Rust frontend, Python engine, tokenizer/template loading, and OpenAI HTTP serving all started successfully.
- Non-streaming output preserved generated text as content and prepended exactly one marker: `content = "<think>hello world"`, `reasoning = null`.
- Streaming reconstruction produced `content = "<think>hello world"`, empty reasoning, one `<think>` prefix, and `finish_reason = "stop"`.

A model-quality evaluation is not applicable because this change does not affect model execution, logits, or sampling; the full-model run validates parser integration and HTTP behavior.

## AI assistance

AI assistance was used to analyze the parser architecture, implement the change, and run validation. The human submitter reviewed the changed lines and is responsible for understanding and defending the change end-to-end.

### CODE REVIEW

**Comment by 982945902** [comment]
The `pre-run-check` is currently blocked by the first-time-contributor gate because this account has no merged vLLM PRs yet and cannot apply repository labels.

Could a maintainer add the `verified` label so the pre-commit workflow can run while this remains a draft?

Local validation completed:

- `cargo test --manifest-path rust/Cargo.toml -p vllm-parser --lib` — 420 passed
- `cargo test --manifest-path rust/Cargo.toml -p vllm-chat --lib` — 266 passed
- `cargo clippy --manifest-path rust/Cargo.toml -p vllm-parser -p vllm-chat --all-targets -- -D warnings` — passed
- pre-commit hooks on all changed files — passed

The draft will remain marked as such until the human submitter has reviewed every changed line, per the repository's AI-assisted contribution policy.

**Comment by 982945902** [comment]
Full-model serving regression completed on August 17, 2026 and added to the PR description. Using the locally built PR Rust frontend with a real 7B CPU engine, non-streaming and streaming responses both produced `<think>hello world`, emitted the prefix exactly once, kept reasoning empty, and completed normally. MiniMax M2.5 itself was not downloaded because its index reports approximately 434 GiB across 125 shards; parser selection was explicit, so the real-engine HTTP integration path is still exercised end to end.

**Comment by 982945902** [comment]
Rebased cleanly onto the latest `main`; there was no file-level merge conflict.

Validation after the rebase:

- `cargo test --manifest-path rust/Cargo.toml -p vllm-parser --lib` — passed
- `cargo test --manifest-path rust/Cargo.toml -p vllm-chat --lib` — 269 passed
- `cargo clippy --manifest-path rust/Cargo.toml -p vllm-parser -p vllm-chat --all-targets -- -D warnings` — passed
- pre-commit hooks on all PR files — passed

**Comment by 982945902** [comment]
Rebased onto the latest `main`, fixed the commit authorship/sign-off so DCO now passes, and added explicit coverage/documentation for empty generated streams.

Validation on the rebased commit:

- `cargo test --manifest-path rust/Cargo.toml -p vllm-parser --lib` — 427 passed
- `cargo test --manifest-path rust/Cargo.toml -p vllm-chat --lib` — 276 passed
- `cargo clippy --manifest-path rust/Cargo.toml -p vllm-parser -p vllm-chat --all-targets -- -D warnings` — passed
- changed-file pre-commit hooks — passed

The remaining failed `pre-run-check` is the first-time-contributor gate. Could a maintainer add `verified` or `ready` to allow CI to run?

**Comment by 982945902** [comment]
Rebased onto current `main` (`687db59744`) and resolved the parser-name snapshot conflict.

Validation on the rebased head:
- `vllm-parser`: 435 passed
- `vllm-chat`: 279 passed
- Clippy with `-D warnings`: passed
- changed-file pre-commit hooks: passed

**Comment by 982945902** [comment]
Rebased onto current `main` (`d2906091bf`) and resolved the reasoning-parser registration/test conflicts. I also updated the parser to the current `DecodedText` API while preserving token attributions.

Validation on the rebased head (`10939d16e9`):
- `cargo test --manifest-path rust/Cargo.toml -p vllm-parser --lib` — 459 passed
- `cargo test --manifest-path rust/Cargo.toml -p vllm-chat --lib` — 293 passed
- `cargo clippy --manifest-path rust/Cargo.toml -p vllm-parser -p vllm-chat --all-targets -- -D warnings` — passed
- changed-file pre-commit hooks — passed
